pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -56,27 +56,30 @@ type ClientConfig interface {
 	Namespace() (string, bool, error)
 }
 
+type PersistAuthProviderConfigForUser func(user string) restclient.AuthProviderConfigPersister
+
 // DirectClientConfig is a ClientConfig interface that is backed by a clientcmdapi.Config, options overrides, and an optional fallbackReader for auth information
 type DirectClientConfig struct {
-	config         clientcmdapi.Config
-	contextName    string
-	overrides      *ConfigOverrides
-	fallbackReader io.Reader
+	config                    clientcmdapi.Config
+	contextName               string
+	overrides                 *ConfigOverrides
+	fallbackReader            io.Reader
+	persistAuthProviderConfig PersistAuthProviderConfigForUser
```

Comment:
@deads2k you mean strongly-typed persisters? How would one do that in Go, given the lack of generics, templating, etc.